app/state/middlewares/persistence/__tests__/persistence.test.js

```diff
@@ -99,15 +99,17 @@ describe('persistence middleware', () => {
       expectedReturnedAction = { someAction: 'someAction' }
       next = jest.fn(() => expectedReturnedAction)
   
+      console.log('this is ok')
       const returnedAction = await callPersistenceMiddleware(action)
+      console.log('something went wrong after await')
   
       it('should not dispatch any action', () => {
         expect(store.dispatch).not.toHaveBeenCalled()
       })
   
       it('should call propagate the action to next middleware', () => {
         expect(store.next.toHaveBeenCalledWith(action))
-        expect(returnedAction).toEqual(returnedAction)
+        expect(returnedAction).toEqual("?")
```

Comment:
@imranariffin this will actually pass. I think it's related to async/await.

Reply:
I think I found the cause of the issue: it's caused by using async/await within foreach

I added two possible solutions to this in two commits separate commits: cd31c56 & bf19caa

ref: https://github.com/babel/babel/issues/909

@cglotr